pkg/metrics/percentile/gk_array.go

```diff
@@ -194,6 +194,77 @@ func (s GKArray) interpolatedQuantile(q float64) float64 {
 	return weightBelow*s.Entries[indexBelow].V + weightAbove*s.Entries[indexAbove].V
 }
 
+// Quantiles accepts a sorted slice of quantile values and returns the epsilon estimates of
+// elements at those quantiles.
+func (s GKArray) Quantiles(qValues []float64) []float64 {
+	quantiles := make([]float64, 0, len(qValues))
+	if s.Count == 0 {
+		for _ = range qValues {
+			quantiles = append(quantiles, math.NaN())
+		}
+		return quantiles
+	}
+
+	// This shouldn't happen but checking just in case.
+	if len(s.Incoming) > 0 {
+		s = s.compressWithIncoming(nil)
+	}
+
+	// When there are only a few values just call interpolatedQuantile
+	// for each quantile.
+	if s.Count < int64(1/EPSILON) {
+		for _, q := range qValues {
+			if q < 0 || q > 1 {
+				quantiles = append(quantiles, math.NaN())
+			} else {
+				quantiles = append(quantiles, s.interpolatedQuantile(q))
```

Comment:
minor nit: typical golang style would have this as
```go
if q < 0 || q > 1 {
    quantiles = append(quantiles, math.NaN())
    continue
}

quantiles = append(quantiles, s.interpolatedQuantile(q))

```

Reply:
👌